plugins/org.yakindu.sct.generator.c/src/org/yakindu/sct/generator/c/StateConfVectorIndexCalculator.xtend

```diff
@@ -0,0 +1,46 @@
+package org.yakindu.sct.generator.c
+
+import com.google.inject.Inject
+import java.util.HashMap
+import java.util.Map
+import org.yakindu.sct.model.sexec.ExecutionState
+import org.yakindu.sct.model.sexec.naming.INamingService
+
+class StateConfVectorIndexCalculator {
+	@Inject extension INamingService
+	protected Map<ExecutionState, Integer> indexMap
+	protected Map<ExecutionState, String> defineMap
+	
+	def Integer getIndex(ExecutionState it) {
```

Comment:
I would remove the Map. Its just a call to stateVector.offset, there is no need to cache the result for this. Then the IndexCalculator can be used stateless and as a singleton.